symengine/polynomial.cpp

```diff
@@ -160,16 +160,14 @@ integer_class UnivariateIntPolynomial::eval(const integer_class &x) const {
 integer_class UnivariateIntPolynomial::eval_bit(const int &x) const {
 
     unsigned int last_deg = dict_.rbegin()->first;
-    integer_class result(0), two_pow;
+    integer_class result(0);
 
     for (auto it = dict_.rbegin(); it != dict_.rend(); ++it) {
 
-        two_pow = integer_class(1) << x * (last_deg - (*it).first);
+        result = (*it).second + (result << x * (last_deg - (*it).first));
```

Comment:
This is a good use case to check whether https://github.com/symengine/symengine/issues/478#issuecomment-153172972 would have a significant performance improvement when `-DINTEGER_CLASS=gmp` is given